Add BCubeF745v2 board (#32753)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/BCubeF745v2/README.md

## Where to Buy

Available from [Aliexpress](https://it.aliexpress.com/item/1005008279741542.html)

## Specifications

- **Processor**
  - STM32F745 32-bit processor
  - 1MB Flash
  - 16Mb Blackbox flash (M25P16 Dataflash on SPI1)
  - AT7456E OSD (on SPI2)

- **Sensors**
  - Invensense ICM-42688-P IMU (on SPI4)
  - DPS310 Barometer (on I2C1)

- **Interfaces**
  - USB Type-C
  - 6 PWM Outputs (5 with DShot and 4 with Bi-Dir DShot support)
  - 6 UARTs / USARTs
  - 1 I2C buses (I2C1)
  - Current, Voltage
  - Buzzer output
  - Status LED

- **Power**
  - Input voltage: 3-8S LiPo (11.1V - 33.6V)
  - Dual BEC: 5V/2.5A and 10V/2A

- **Dimensions**
  - Size: 36 x 36 mm
  - Mounting holes: 30.5 x 30.5 mm (M4)
  - Weight: 8.8 g

## Pinout

Please refer to the silk screen markings on the board for pad definitions.

## UART Mapping

The UARTs are configured and mapped as follows:

| Port   | UART   | Protocol   | TX DMA | RX DMA | Notes |
|--------|--------|------------|--------|--------|-------|
| 0      | OTG1   | MAVLink2   | ✘      | ✘      | USB |
| 1      | USART1 | MAVLink2   | ✘      | ✘      | |
| 2      | USART2 | RCIN       | ✔      | ✔      | Default RC input |

> | 3      | USART3 | GPS1       | ✔      | ✔      | |

Reviewer: ```suggestion
| 3      | USART3 | USER   | ✔      | ✔      | |
```

Author: done

Reviewer: but you did not change its protocol from GPS (defaults to this automatically) to NONE

| 4      | UART4  | GPS2       | ✘      | ✘      | |
| 6      | USART6 | Display    | ✔      | ✔      | |
| 7      | UART7  | ESC Telemetry | ✘ | ✘      | RX only |

> *Note: Serial 6 is conventionally used for RC input, defaulted to protocol 23 (RCIN).*

Reviewer: remove above line

Author: done

## RC Input

> By default, the RC input is mapped to USART2 (RX pad). This interface supports all ArduPilot compatible RC systems including SBUS, CRSF, and FPort. USART2 has DMA enabled, satisfying the requirements for protocols like CRSF/ELRS and SRXL2.

Reviewer: ```suggestion
By default, the RC input is mapped to USART2 (RX pad). This interface supports all ArduPilot compatible RC systems, except PPM, including SBUS, CRSF, and FPort. USART2 has DMA enabled, satisfying the requirements for protocols like CRSF/ELRS and SRXL2.
```

Author: done

## PWM Outputs

The B-CUBE F745 V2 controller supports 6 PWM outputs. The outputs support normal PWM output formats.
Outputs 1-4 support DShot and Bi-Directional DShot. Output 6 supports DShot, but not Bi-Directional DShot.

> Output 5 does not have DMA assigned, so it does not support DShot.

Reviewer: ```suggestion
Output 5 does not have DMA assigned, so it does not support DShot. Output 7 is defaulted to serial LED, but can be used for PWM or DShot by changing its :ref:`SERVO7_FUNCTION <SERVO7_FUNCTION >`
```

Author: done

The PWM outputs are grouped by timers:

- Outputs 1, 4 are in group1 (TIM3)
- Outputs 2, 3 are in group2 (TIM1)
- Output 5 is in group3 (TIM8)

> - Output 6 is in group4 (TIM5)

Reviewer: ```suggestion
- Output 6 is in group4 (TIM5)
- Output 7 (LED) in group5 (TIM4)
```

Author: done

Channels within the same group need to use the same output rate. If any channel in a group uses DShot, then all channels in the group need to use DShot.

## GPIOs

The hardware features the following custom GPIO mappings:

| Pin | GPIO Number |
|--------------|-------------|
| PWM(1)       | 50          |
| PWM(2)       | 51          |
| PWM(3)       | 52          |
| PWM(4)       | 53          |
| PWM(5)       | 54          |
| PWM(6)       | 55          |
| ALARM/BUZZER | 77          |

## OSD Support

> On-board analog OSD (using the AT7456E) is supported and enabled by default.

Reviewer: ```suggestion
On-board analog OSD (using the AT7456E) is supported and enabled by default. Simultaneous HD VTX (DisplayPort) operation on SERIAL6 is also enabled.
```

Author: done

## Compass

> The B-CUBE F745 V2 does not have a built-in compass. However, it can connect robustly to an external compass module (e.g. QMC5883L) via the unpopulated pads for I2C1 (PB6=SCL, PB7=SDA).

Reviewer: ```suggestion
The B-CUBE F745 V2 does not have a built-in compass. However, an exteranl compass can connected using the SCL and SDA pads.
```

Author: done

> The B-CUBE F745 V2 does not have a built-in compass. However, an exteranl compass can be connected using the SCL and SDA pads.

Reviewer: sorry, my miss-spelling

```suggestion
The B-CUBE F745 V2 does not have a built-in compass. However, an external compass can be connected using the SCL and SDA pads.
```

Author: My mistake too! Fixed.
